pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1516,6 +1483,48 @@ $func$
 LANGUAGE PLPGSQL VOLATILE;
 GRANT EXECUTE ON FUNCTION SCHEMA_CATALOG.create_metric_view(text) TO prom_writer;
 
+CREATE OR REPLACE FUNCTION SCHEMA_CATALOG.delete_series_from_metric(series_ids bigint[], name text)
+RETURNS BIGINT
+LANGUAGE PLPGSQL
+AS
+$$
+DECLARE
+    metric_table name;
+    delete_stmt text;
+    delete_query text;
+    rows_affected bigint;
+    num_chunks_touched bigint := 0;
+BEGIN
+    SELECT table_name INTO metric_table FROM SCHEMA_CATALOG.metric m WHERE m.metric_name=name;
+    IF SCHEMA_CATALOG.is_timescaledb_installed() THEN
+        FOR delete_stmt IN
+            SELECT FORMAT('DELETE FROM %1$I.%2$I WHERE series_id = ANY($1)', schema_name, table_name)
+            FROM (
+                SELECT (COALESCE(chc, ch)).* FROM pg_class c
+                    INNER JOIN pg_namespace n ON c.relnamespace = n.oid
+                    INNER JOIN _timescaledb_catalog.chunk ch ON (ch.schema_name, ch.table_name) = (n.nspname, c.relname)
+                    LEFT JOIN _timescaledb_catalog.chunk chc ON ch.compressed_chunk_id = chc.id
+                WHERE c.oid IN (SELECT show_chunks(format('%I.%I','SCHEMA_DATA', metric_table))::oid)
+                ) a
+        LOOP
+            EXECUTE delete_stmt USING series_ids;
+            GET DIAGNOSTICS rows_affected = ROW_COUNT;
+            num_chunks_touched = num_chunks_touched + rows_affected;
```

Comment:
This is no longer num_chunks_touched, but rather a row count...should we call it that to avoid being confusing with naming?

Reply:
sure, but won't the row_count denote the number of rows that were updated (or deleted)?

Reply:
(though it's only counting compressed rows once, so it's a bit misleading if we call it that too)

Reply:
yes, it should be changed to `num_rows_affected`. I will update once this PR is reviewed so that it comes along with other changes if required.

Reply:
yes I like `num_rows_affected` or `num_rows_deleted`

Reply:
Do we care about the compressed rows thing? We could do a returning with our stats column that has the number of rows and use that to get the actual number of rows deleted?

Reply:
@cevian thoughts?